# Tugas Akhir
# Chihago Crimes Dataset

##### Anggota Kelompok:
1. Aditya Arya Hendrady (1506722733)
2. Arfi Renaldi (1506722872)
3. Harry Akbar Ali Munir (1506689080)
4. Rifki Adrian (1506726946)

###### Kelas A
###### Faculty of Computer Science, University of Indonesia

## 1) Environment Preparation

### 1.a) Library installation

In [ ]:
import pandas as pd
import numpy as np
import random
import seaborn as sns
import matplotlib.pyplot as plt

from sklearn import preprocessing
from sklearn.tree import DecisionTreeRegressor
from sklearn.cross_validation import train_test_split
from sklearn import metrics
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import mean_squared_error
from sklearn.preprocessing import LabelEncoder

## 2) Data Preparation

In [10]:
df = pd.read_csv("Crimes_-_2001_to_present.csv")

print("Jumlah row: {}".format(df.shape[0]))
print("Jumlah feature: {}".format(df.shape[1]))
df.head()

Jumlah feature: 6582553
Jumlah row: 22


ID Case Number                    Date                   Block  IUCR  \
0  10000092    HY189866  03/18/2015 07:44:00 PM         047XX W OHIO ST  041A   
1  10000094    HY190059  03/18/2015 11:00:00 PM  066XX S MARSHFIELD AVE  4625   
2  10000095    HY190052  03/18/2015 10:45:00 PM   044XX S LAKE PARK AVE  0486   
3  10000096    HY190054  03/18/2015 10:30:00 PM    051XX S MICHIGAN AVE  0460   
4  10000097    HY189976  03/18/2015 09:00:00 PM        047XX W ADAMS ST  031A   

    Primary Type              Description Location Description Arrest  \
0        BATTERY      AGGRAVATED: HANDGUN               STREET  False   
1  OTHER OFFENSE         PAROLE VIOLATION               STREET   True   
2        BATTERY  DOMESTIC BATTERY SIMPLE            APARTMENT  False   
3        BATTERY                   SIMPLE            APARTMENT  False   
4        ROBBERY           ARMED: HANDGUN             SIDEWALK  False   

  Domestic              ...                Ward  Community Area  FBI Code  \
0    False              ...                28.0            25.0       04B   
1    False              ...                15.0            67.0        26   
2     True              ...                 4.0            39.0       08B   
3    False              ...                 3.0            40.0       08B   
4    False              ...                28.0            25.0        03   

   X Coordinate Y Coordinate  Year              Updated On   Latitude  \
0     1144606.0    1903566.0  2015  02/10/2018 03:50:01 PM  41.891399   
1     1166468.0    1860715.0  2015  02/10/2018 03:50:01 PM  41.773372   
2     1185075.0    1875622.0  2015  02/10/2018 03:50:01 PM  41.813861   
3     1178033.0    1870804.0  2015  02/10/2018 03:50:01 PM  41.800802   
4     1144920.0    1898709.0  2015  02/10/2018 03:50:01 PM  41.878065   

   Longitude                       Location  
0 -87.744385  (41.891398861, -87.744384567)  
1 -87.665319  (41.773371528, -87.665319468)  
2 -87.596643   (41.81386068, -87.596642837)  
3 -87.622619  (41.800802415, -87.622619343)  
4 -87.743354  (41.878064761, -87.743354013)  

[5 rows x 22 columns]

### 2.a) Data Cleansing

#### Sampling

Sampling diperlukan karena terdapat jumlah baris yang sangat banyak (6582553 baris) sehingga akan menyebabkan proses komputasi yang besar

In [12]:
import numpy as np
import random

df_sample = df.sample(frac=0.3)
print("Jumlah row: {}".format(df_sample.shape[0]))
print("Jumlah feature: {}".format(df_sample.shape[1]))
df

Jumlah row: 197476
Jumlah feature: 22


ID Case Number                    Date  \
1897975   1429089     G136231  03/09/2001 01:45:00 PM   
2860483   8176822    HT408288  07/21/2011 04:00:00 AM   
5360149   4736611    HM344723  05/11/2006 05:30:00 PM   
2535169   7571610    HS375443  06/21/2010 08:11:00 PM   
5246203   4519519    HM106764  01/04/2006 05:45:00 PM   
1119850   3910630    HL286742  04/10/2005 12:01:00 AM   
1154195   7794032    HS602001  10/24/2010 02:20:00 PM   
5698566   5526053    HN322174  05/04/2007 01:20:00 PM   
5641320   5378128    HN227584  03/13/2007 12:00:00 PM   
18622    10035495    HY224809  04/16/2015 07:59:00 PM   
6504629   7187409    HR598753  10/20/2009 08:00:00 PM   
5903241   5954188    HN742723  11/30/2007 05:00:00 PM   
1723177   6228094    HP312891  05/02/2008 12:15:00 PM   
5020237   4002882    HL362379  05/17/2005 07:24:54 PM   
4013316   2396765    HH707483  10/11/2002 12:00:00 PM   
2422445   7399823    HS201637  03/10/2010 10:00:00 PM   
4774087   3504569    HK580441  08/25/2004 09:40:00 AM   
6237186   6656691    HP728021  12/11/2008 08:40:00 AM   
1498876   8781170    HV454656  08/31/2012 03:15:00 AM   
499321   10951301    JA263488  05/13/2017 11:00:00 PM   
3881430   2182277    HH428211  06/08/2002 06:40:00 PM   
5370537   4758652    HM362586  05/20/2006 10:34:00 AM   
5490552   5020432    HM610383  09/19/2006 12:30:00 PM   
4980090   3897954    HL270653  04/02/2005 06:00:38 PM   
1532999   5155676    HM737926  11/24/2006 01:39:00 PM   
4541373   3132476    HK119736  01/11/2004 08:20:00 PM   
1943554   1487370     G223539  04/19/2001 09:00:00 PM   
6267733   6716587    HR132862  01/22/2009 12:00:00 AM   
2366688   7308060    HS112321  01/09/2010 12:45:00 PM   
4631367   3274456    HK129510  01/16/2004 08:04:00 PM   
...           ...         ...                     ...   
2861959   8178563    HT413801  07/24/2011 07:43:00 PM   
585521   11092705    JA437853  09/11/2017 05:00:00 PM   
3669108   9829772    HX479246  09/10/2014 10:42:00 AM   
2839904   8145339    HT380124  07/04/2011 08:20:00 PM   
749387    2175539    HH431365  06/09/2002 08:00:00 PM   
4095252   2526695    HJ102282  01/02/2003 09:21:56 AM   
6180656   6545814    HP594651  09/26/2008 04:00:00 PM   
6131409   6445607    HP519463  08/17/2008 11:35:00 PM   
6447990   7080593    HR488976  08/17/2009 05:15:00 PM   
3972737   2331126    HH611782  08/29/2002 11:45:00 PM   
2426247   7405025    HS197748  03/08/2010 03:45:00 PM   
3448927   9378755    HW522094  07/01/2013 09:00:00 AM   
3879772  10506966    HZ248847  05/02/2016 08:00:00 AM   
1172523   7842197    HS653819  12/10/2010 06:00:00 AM   
23557    10044866    HY233375  04/23/2015 03:00:00 PM   
6543697   7261488    HR665566  11/29/2009 04:15:00 AM   
4630545   3273080    HK298281  04/12/2004 07:59:12 PM   
5072520   4135602    HL391195  05/31/2005 11:15:00 PM   
3568342   9663892    HX314080  06/22/2014 12:00:00 AM   
725602   11242920    JB168247  02/28/2018 08:29:00 AM   
374788   10695287    HZ450224  09/24/2016 03:20:00 PM   
646498    5929863    HN730133  09/27/2005 09:00:00 AM   
5923182   6000788    HP106994  01/05/2008 12:15:00 AM   
623826   11154729    JA519134  11/19/2017 08:00:00 PM   
4176753   2644948    HJ256171  03/23/2003 01:00:00 PM   
4788045   3526744    HK606016  09/06/2004 12:45:35 PM   
5290857   4602854    HM180438  02/14/2006 10:55:00 AM   
4865093   3667256    HK768006  11/22/2004 11:00:00 PM   
765101    2202632    HH453590  06/19/2002 09:15:00 PM   
2845864   8154423    HT389169  07/10/2011 12:30:00 AM   

                                       Block  IUCR  \
1897975                      030XX S KING DR  0460   
2860483               098XX S VANDERPOEL AVE  5002   
5360149                     001XX N STATE ST  0860   
2535169                103XX S GREEN BAY AVE  1750   
5246203                    113XX S PEORIA ST  0470   
1119850                       003XX N MAY ST  0820   
1154195                 024XX N CAMPBELL AVE  2825   
5698566                060XX S HERMITAGE AVE  18

### 2.b) Feature Extraction

### 2.c) Data Analysis

## 3) Accuracy Testing

### 3.a) Test a few machine learning algorithm with your dataset